app/models/ems_cluster.rb

```diff
@@ -107,8 +109,8 @@ def total_vcpus
   # Relationship methods
   #
 
-  alias_method :storages,               :all_storages
-  alias_method :datastores,             :all_storages    # Used by web-services to return datastores as the property name
+  has_many :storages, -> { distinct }, :through => :hosts
```

Comment:
please move up to where you added the has_many above

think it needs to be above the aggregation mixin. maybe?